modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/ExchangeServiceImpl.java

```diff
@@ -197,8 +198,10 @@ protected void onMessage(UUID nodeId, QueryBatchAcknowledgeMessage msg) {
             try {
                 outbox.onAcknowledge(nodeId, msg.batchId());
             }
-            catch (Throwable t) {
-                outbox.onError(t);
+            catch (Throwable e) {
+                outbox.onError(e);
+
+                throw new IgniteException("Unexpected exception", e);
             }
         }
         else if (log.isDebugEnabled()) {
```

Comment:
seems like you've missed one more place. PLease see `protected void onMessage(UUID nodeId, QueryBatchMessage msg)`